src/eos-util.c

```diff
@@ -0,0 +1,423 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
+ *
+ * Copyright © 2013 Collabora Ltd.
+ * Copyright 2016 Kinvolk GmbH
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ * Authors: Vivek Dasmohapatra <vivek@etla.org>
+ *          Krzesimir Nowak <krzesimir@kinvolk.io>
+ */
+
+#include "eos-util.h"
+
+#include <libsoup/soup.h>
+
+#include <string.h>
+
+static const GDBusErrorEntry eos_updater_error_entries[] = {
+  { EOS_UPDATER_ERROR_WRONG_STATE, "com.endlessm.Updater.Error.WrongState" },
+  { EOS_UPDATER_ERROR_LAN_DISCOVERY_ERROR, "com.endlessm.Updater.Error.LANDiscoveryError" },
+  { EOS_UPDATER_ERROR_WRONG_CONFIGURATION, "com.endlessm.Updater.Error.WrongConfiguration" }
+};
+
+/* Ensure that every error code has an associated D-Bus error name */
+G_STATIC_ASSERT (G_N_ELEMENTS (eos_updater_error_entries) == EOS_UPDATER_N_ERRORS);
+
+GQuark
+eos_updater_error_quark (void)
+{
+  static volatile gsize quark_volatile = 0;
+  g_dbus_error_register_error_domain ("eos-updater-error-quark",
+                                      &quark_volatile,
+                                      eos_updater_error_entries,
+                                      G_N_ELEMENTS (eos_updater_error_entries));
+  return (GQuark) quark_volatile;
+}
+
+static const gchar * state_str[] = {
+   "None",
+   "Ready",
+   "Error",
+   "Polling",
+   "UpdateAvailable",
+   "Fetching",
+   "UpdateReady",
+   "ApplyUpdate",
+   "UpdateApplied" };
```

Comment:
I didn't write it (it comes from [the old eos-updater-util.h](https://github.com/endlessm/eos-updater/blob/master/src/eos-updater-util.c#L54)), but sure, I'll fix it.